moped-editor/src/app/views/forms/NewProject.jsx

```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import { useForm, Controller } from "react-hook-form";
+import { TextField, Button, Grid, Icon, InputLabel, MenuItem, Select } from "@material-ui/core";
+import { Breadcrumb } from "matx";
+import { gql, useMutation, useQuery } from "@apollo/client";
+
+const NewProject = () => {
+
+  const { register, handleSubmit, setValue, control } = useForm();
+
+  const PHASES_QUERY = gql`
+  query Phases {
+    moped_phases(order_by: {phase_name: asc}) {
+      phase_name
+    }
+  }
+`;
+
+const STATUS_QUERY = gql`
+  query Status {
+    moped_status(order_by: {status_name: asc}) {
+      status_name
+    }
+  }
+`;
+
+const FISCAL_QUERY = gql`
+  query Fiscal {
+    moped_city_fiscal_years(order_by: {fiscal_year_value: asc}) {
+      fiscal_year_value
+    }
+  }
+`;
+
+const PRIORITY_QUERY = gql`
+  query Priority {
+    moped_proj_phases(order_by: {phase_priority: asc}) {
+      phase_priority
+    }
+  }
+`;
+
+  const addNewProject = gql `
+  mutation MyMutation($project_name: String!="", $project_description: String!="", $current_phase: String!="", $project_priority: Int!="", $current_status: String!="", $eCapris_id: bpchar!="", $fiscal_year: String!="", $capitally_funded: Boolean!="", $start_date: date="") {
+    insert_moped_project(objects: {project_name: $project_name, project_description: $project_description, project_priority: $project_priority, current_phase: $current_phase, current_status: $current_status, eCapris_id: $eCapris_id, fiscal_year: $fiscal_year, capitally_funded: $capitally_funded, start_date: $start_date }) {
+      affected_rows
+      returning {
+        project_name
+        project_description
+        project_priority
+        current_phase
+        current_status
+        eCapris_id
+        fiscal_year
+        capitally_funded
+        start_date
+      }
+    }
+  }    
+ `;
+
+  const [addProject] = useMutation(addNewProject);
+   
+  const onSubmit = (data, e) => {
+    e.target.reset(data); // reset after form submit
+    let project_name=data.newProject;
+    let project_description=data.ProjDesc;
+    let eCapris_id=data.eCaprisId;
+    let capitally_funded=data.capitalFunded;
+    let start_date=data.date;
+    let current_phase=data.Phase;
+    let project_priority=data.Priority;
+    let current_status=data.Status;
+    let fiscal_year=data.FiscalYear;
+    addProject({variables: {project_name, project_description, eCapris_id, current_phase, current_status, project_priority, fiscal_year, capitally_funded, start_date}});    
```

Comment:
Why not just pass the data.* variables directly to the addProject function? ie:

```
addProject({variables: {data.newProject, data.ProjDesc, data.eCaprisId, ...}});
```

Reply:
One additional option that might be worth exploring here or down the road at some point: https://hasura.io/docs/1.0/graphql/core/mutations/insert.html#insert-a-single-object
(the second example called "Using variables:")

If the names of the form fields are updated to the format that Hasura is expecting, it might be possible just to pass `data` in this mutation. I know it breaks JS convention with the snake case keys in `data`, but it might be cool to have them map directly. Just a thought and not sure how possible that is. Eager to take a closer look at all of this! Cool stuff!

Reply:
I checked out the link and it is definitely worth looking into. I'll try both suggestions and see what happens.